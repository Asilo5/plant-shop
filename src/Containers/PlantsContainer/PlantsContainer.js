import React from 'react';
import './PlantsContainer.scss';
import Plant from '../../Components/Plant/Plant';
import { connect } from 'react-redux';

export const PlantsContainer = ({ allPlants }) => {
   let plantsList = allPlants.map((plant) => {
     return <Plant key={plant._id} {...plant} />
   });

    return(
      <section>
          <div className='plants-header'>
            <h2>Our Plants</h2>
          </div>
          <div className='plant-list'>
            {plantsList}
          </div>
      </section>
    )
}

export const mapStateToProps = (state) => ({
    allPlants: state.allPlants
})

export default connect(mapStateToProps)(PlantsContainer);