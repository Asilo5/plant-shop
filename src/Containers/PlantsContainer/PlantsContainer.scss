.plants-header {
    background-image: url('https://images.unsplash.com/photo-1483794344563-d27a8d18014e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60');
    background-size: cover;
    background-position: center;
    height: 40vh;
    h2 {
        text-align: center;
        position: relative;
        top: 133px;
        font-size: 5em;
    }
}

.plant-list {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
    padding: 30px;
    background: white;
}