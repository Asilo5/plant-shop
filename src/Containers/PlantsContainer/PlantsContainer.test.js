import React from 'react';
import { shallow } from 'enzyme';
import {PlantsContainer, mapStateToProps } from './PlantsContainer';

describe('PlantsContainer container', () => {

    describe('PlantsContainer', () => {
        it('should match snapshot', () => {
            let mockAllPlants = [
                {
                    "_id": "5e16b29d88553086edcddc53",
                    "name": "Aloe Queen",
                    "description": "Healing plant",
                    "care": "Lots of Love",
                    "price": 20,
                    "image": "https://target.scene7.com/is/image/Target/GUEST_be7e6313-15ac-4e1c-94b8-8af5595d6cb6?wid=488&hei=488&fmt=pjpeg",
                    "__v": 0
                },
            ];
    
            let wrapper = shallow(<PlantsContainer allPlants={mockAllPlants} />);
    
            expect(wrapper).toMatchSnapshot();
        })
    })

    describe('mapStateToProps', () => {
        it('should return all state with their properties', () => {
            const mockState = {
                "allPlants": [
                    {
                          "__v": 0,
                          "_id": "5e16b29d88553086edcddc53",
                          "care": "Lots of Love",
                          "description": "Healing plant",
                          "image": "https://target.scene7.com/is/image/Target/GUEST_be7e6313-15ac-4e1c-94b8-8af5595d6cb6?wid=488&hei=488&fmt=pjpeg",
                          "name": "Aloe Queen",
                          "price": 20,
                     },
                      ]
            }

            const expected = {
                "allPlants": [
                    {
                          "__v": 0,
                          "_id": "5e16b29d88553086edcddc53",
                          "care": "Lots of Love",
                          "description": "Healing plant",
                          "image": "https://target.scene7.com/is/image/Target/GUEST_be7e6313-15ac-4e1c-94b8-8af5595d6cb6?wid=488&hei=488&fmt=pjpeg",
                          "name": "Aloe Queen",
                          "price": 20,
                     },
                      ]
            }

            const mappedProps = mapStateToProps(mockState);
            expect(mappedProps).toEqual(expected);
        })
    })
})
