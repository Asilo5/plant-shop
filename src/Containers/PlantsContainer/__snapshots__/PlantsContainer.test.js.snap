// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantsContainer container PlantsContainer should match snapshot 1`] = `
<section>
  <div
    className="plants-header"
  >
    <h2>
      Our Plants
    </h2>
  </div>
  <div
    className="plant-list"
  >
    <Plant
      __v={0}
      _id="5e16b29d88553086edcddc53"
      care="Lots of Love"
      description="Healing plant"
      image="https://target.scene7.com/is/image/Target/GUEST_be7e6313-15ac-4e1c-94b8-8af5595d6cb6?wid=488&hei=488&fmt=pjpeg"
      key="5e16b29d88553086edcddc53"
      name="Aloe Queen"
      price={20}
    />
  </div>
</section>
`;
